### DISCUSSION TITLE
how to handle MTD which powers off

### DISCUSSION THREAD

**Discussion by visuphi** [openthread/openthread#8673] [Q&A] [answered]
We have a node with a few sensors which we configure as MTD. The node switches on, reads the sensors and then sends the data via CoAP/Openthread. The node then turns everything off with a load switch and sleeps for 10 minutes to keep it ultra low power.

As I understand, this does not correspond to a SED, since we switch everything off.  After powering on again, the node then goes through the procedure again with parent request, etc.

questions:

1. Does it have any (negative) effect on the thread network if our node joins the network every 10 minutes (with parent request, child ID request, etc.)? 
2.  Does it make sense to set the child time out short, so that the node disappears from the child table?
3. is there anything else we should consider when proceeding in this way (cut off the power)?

**Comment by jwhui** [comment, marked as answer]
> Does it have any (negative) effect on the thread network if our node joins the network every 10 minutes (with parent request, child ID request, etc.)?

The overall effect to the network should be minimal. The network will maintain some stale state (e.g. parent maintains entry in child table, address cache mappings from IPv6 address to RLOC, etc.) but if there is no communication to/from the device, that should not have much affect much. If the child is initiating the communication, then the address cache entries should get updated with the child's initial communications after joining again.

> Does it make sense to set the child time out short, so that the node disappears from the child table?

Setting the child timeout small is useful if the child is not expected to stay attached for very long. It helps clean up state that I mentioned above more quickly.

> is there anything else we should consider when proceeding in this way (cut off the power)?

Attaching to the network is a fairly expensive operation, since it involves having the child keep its receiver on when looking for neighboring parents. I believe it would be more energy efficient to have the device maintain its connection to the parent but simply go into a ultra low power state (enough to maintain a wakeup timer). Performing a full reattach only makes sense if you expect some device mobility, where the existing parent-child link is not likely to be valid any more.

Hope that helps.
